lib/kubernetes-deploy/kubectl.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+require 'kubeclient'
+
+module KubernetesDeploy
+  module Kubectl
+    class KubectlError < FatalDeploymentError; end
+
+    def run_kubectl(*args, namespace:, context:)
+      args = args.unshift("kubectl")
+      args.push("--namespace=#{namespace}") if namespace
+      args.push("--context=#{context}")     if namespace
+
+      KubernetesDeploy.logger.debug Shellwords.join(args)
+      out, err, st = Open3.capture3(*args)
+      KubernetesDeploy.logger.debug(out.shellescape)
+      unless st.success?
```

Comment:
the version in kubernetes resource logs as well only at the debug level rather than warn. I can add a `log_failure` flag to disable the logging.